### PR TITLE
Speedrank

### PR DESCRIPTION
**Pull request by drgoonic**
Added the "rank" command to allow speedrunners to pull their specific stats in the general categories.

The new command has the form:
rank <playername> <category> <type>
playername = name in the speedrun sheet
category = 3star, 4star, 5star, or Torment (defaults to 4star)
type = overall or no-csb (defaults to overall)

### CODE REVIEW

**Comment by drgoonic** [comment]
Oh I forgot to mention this is intended to resolve #29

File: ffrkbot.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const client = new Commando.Client({
 
 client.on('ready', () => {
   console.log(`Logged in as ${client.user.username}!`);
-  client.user.setGame(',help');
```
Comment (brasstax) [inline ffrkbot.js]: This change isn't related and should be in a separate pull request.

File: utilities/speedrank-utils.js [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+ * @return {String} sheet
+ **/
+function getSheet(category, secondaryCategory) {
+  let version = "Overall";
+  if (secondaryCategory === 'no-csb'){
+    version = 'No CSB';
+  }
+  switch (category) {
+    case '3star':
+      category = 'GL 3* '+version+ ' rankings';
```
Comment (brasstax) [inline utilities/speedrank-utils.js]: For consistency's sake, I'd like for you to use Javascript's [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) to match the rest of the documentation for anything here. For example, instead of:
```javascript
category = 'GL 3* '+version+ ' rankings';
```
Try:

```javascript
`category = GL 3* ${version} rankings;`
```

File: commands/misc/speedrank.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+   **/
+  constructor(client) {
+    super(client, {
+      name: 'rank',
+      group: 'misc',
+      memberName: 'speedrank',
+      description: 'Looks up all the entries for a given speedrunner in a specified leaderboard' +
+      ' from the FFRecordKeeper Discord Speedrun challenge leaderboard' +
+      ' (https://docs.google.com/spreadsheets/d/11gTjAkpm4D3uoxnYCN7ZfbiVnKyi7tmm9Vp9HvTkGpw).',
+      examples: ['rank EverythingIsGravy 5star', 'top 10 no-csb', 'top 5 cod', 'top 3 Maliris'],
```
Comment (brasstax) [inline commands/misc/speedrank.js]: Please take out the `'top'` examples from this and add an example for the rank command without a category, ie `rank littlefiredragon`.

File: utilities/speedrank-utils.js [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+                    contestant.push(checkCell(data.values[row][entryStartPos+1]));
+                  }
+
+                  contestants.push(contestant);
+                  break;
+                }
+              }
+            }
+            const rankTable =
+              outputRankTable(categorySheet, player,
```
Comment (brasstax) [inline utilities/speedrank-utils.js]: Since you're using `player` for your `outputRankTable`, this will use the user's input exactly as requested in the final output. Everything looks right, for example, if you use `rank EverythingIsGravy`, things look fine since that's how they format their own name on the spreadsheet, but if you do `rank EVERYTHINGISGRAVy`, then the results will show:

```
EVERYTHINGISGRAVy - GL 4* Overall rankings
Boss          | Rank | Time  | 
------------------------------
Maliris       | 2    | 7.80  | 
Isgebind      | 2    | 8.02  | 
Tiamat        | 3    | 8.79  | 
Midgardsormr  | 11   | 11.20 | 
Ixion         | 3    | 4.79  | 
Kraken        | 5    | 9.06  | 
Siren         | 5    | 9.27  | 
Hades         | 10   | 13.37 | 
Overall       | 8    | 9.04  | 
```

In line 119, you do a comparison of `cell.toLowerCase` with the player's `toLowerCase`. If it matches, you might want to put the result of `cell.toLowerCase` in its own separate variable, say `playerActualName`, and use that as your input in `outputRankTable`. That way, no matter what the user inputs, the output will always have the correct way the name is displayed on the sheet.

File: commands/misc/speedrank.js (hunk 3 above)
Comment (drgoonic) [reply commands/misc/speedrank.js]: Done, as requested!

File: utilities/speedrank-utils.js (hunk 4 above)
Comment (drgoonic) [reply utilities/speedrank-utils.js]: I did this partially as requested. Rather than use `cell.toLowerCase` I opted for just `cell` as it preserves the formatting the player used for their initial submission to the sheet.

As you suggested above I will break the redundant functions out in a separate commit.
